client/lib/signup/themes.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * External dependencies
+ */
+import shuffle from 'lodash/shuffle';
+
+/**
+ * Internal dependencies
+ */
+import { themes } from 'lib/signup/themes-data';
+
+function filterDesignType( theme ) {
+	return this === theme.design;
+}
+
+function filterVerticals( theme ) {
+	return ! ( -1 === theme.verticals.indexOf( this ) );
+}
+
+function filterDefaultThemes( theme ) {
+	return theme.fallback;
+}
+
+export function getDefaultThemes() {
+	return themes.filter( filterDefaultThemes );
+}
+
+export default function getThemes( vertical, designType ) {
+	let themeSet = themes.filter( filterDesignType, designType ).filter( filterVerticals, vertical );
```

Comment:
This is a very clever use of `filter`! I'm ok with it, but I wonder if we can do it without using `this` (which can often add ambiguity)?

How about:
```
const filterByType = theme => theme.design === designType;
const filterByVertical = theme => includes( theme.verticals, vertical );
let themeSet = themes.filter( filterByType ).filter( filterByVertical );
```